File: project_euler/python/problem2.py
def even_fibonacci_sum():
    nums = [1, 2]
    while True:
        if nums[len(nums) - 1] + nums[len(nums) - 2] > 4000000:
            break
        else:
            nums.append(nums[len(nums) - 1] + nums[len(nums) - 2])
    ans = 0
    for i in nums:
        if i % 2 == 0:
            ans += i
    print(ans)

even_fibonacci_sum()

File: project_euler/go/problem2.go
func evenFibonacciSum(lim int) int {
	nums := []int{1, 2}
	for nums[len(nums)-1]+nums[len(nums)-2] <= lim {
		nums = append(nums, nums[len(nums)-1]+nums[len(nums)-2])
	}
	ans := 0
	for _, num := range nums {
		if num%2 == 0 {
			ans += num
		}
	}
	return ans
}
